### PR TITLE
feat(dedicated): my services add tracking

### PR DESCRIPTION
**Pull request by oalkabouss**
ref: MANAGER-16208

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/my-services-revamp`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-16208
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/billing/src/autoRenew/autorenew.controller.js [hunk 1]
```diff
@@ -86,6 +92,7 @@ export default class AutorenewCtrl {
       this.hideHeaderGuide =
         Object.keys(this.guides.url.my_services).length === 0;
     }
+    this.currentCriteria = JSON.parse(JSON.stringify(this.criteria));
```
Comment (JacquesLarique) [inline packages/manager/modules/billing/src/autoRenew/autorenew.controller.js:95]: Why is this needed ?

File: packages/manager/modules/bm-server-components/src/server-dashboard/dashboard.html [hunk 2]
```diff
@@ -108,7 +108,9 @@
                 go-to-resiliation="$ctrl.goToResiliation()"
                 service-infos="$ctrl.serviceInfos"
                 highlight-engagement="$ctrl.server.shouldReengage"
-                tracking-prefix="{{:: 'dedicated::' + $ctrl.trackingPrefix }}"
+                tracking-prefix="DedicatedServers::dedicated-server::node"
```
Comment (JacquesLarique) [inline packages/manager/modules/bm-server-components/src/server-dashboard/dashboard.html:111]: I'm not sure this is correct, I think this component is used for 'dedicated-servers', 'dedicated-cluster' and 'dedicated-node'

**Review by JacquesLarique — CHANGES_REQUESTED** [review]
Consider using constants instead of putting tracking hit directly in the template

File: packages/manager/modules/billing/src/autoRenew/autorenew.controller.js [hunk 3]
```diff
@@ -198,6 +207,13 @@ export default class AutorenewCtrl {
   }
 
   onCriteriaChange($criteria) {
+    const newCriteria = $criteria.find(
```
Comment (JacquesLarique) [inline packages/manager/modules/billing/src/autoRenew/autorenew.controller.js:210]: Once you found (or not) a new criteria, you need to update `this.currentCriteria` otherwise you'll always find the same changes in the list of criterion

**Review by JacquesLarique — CHANGES_REQUESTED** [review]

File: packages/manager/modules/billing/src/autoRenew/autorenew.controller.js (hunk 3 above)
Comment (oalkabouss) [reply packages/manager/modules/billing/src/autoRenew/autorenew.controller.js:210]: When onListParamChanges is called, the page reloads and onInit is executed with the new values cloned into currentCriteria.

Comment (JacquesLarique) [reply packages/manager/modules/billing/src/autoRenew/autorenew.controller.js:210]: Indeed

**Review by JacquesLarique — APPROVED** [review]

**Review by MaximeBajeux — APPROVED** [review]
